networks/cardano/tokens/dda5fdb1002f7389b33e036b6afee82a8189becb6cba852e8b79b4fb0014df1047454e53.json

```diff
@@ -0,0 +1,19 @@
+{
+    "CMCslug": "genius-yield",
+    "decimals": 6,
+    "mldId": "cardano-GENS-dda5fdb1002f7389b33e036b6afee82a8189becb6cba852e8b79b4fb0014df1047454e53",
+    "orgName": "Genius Yield",
+    "orgUrl": "https://www.geniusyield.co",
+    "tokenAddress": "dda5fdb1002f7389b33e036b6afee82a8189becb6cba852e8b79b4fb0014df1047454e53",
+    "tokenColorTheme": 4,
+    "tokenIcon": "/static/tokens/GENS.svg",
+    "tokenName": "Genius Yield",
```

Comment:
This name is different from [Cexplorer](https://cexplorer.io/asset/asset1266q2ewhgul7jh3xqpvjzqarrepfjuler20akr), it should be `Genius Yield Token`.

Reply:
Corrected.

Reply:
@dongntp can you clarify one point for me:
1. We want to fetch all the data from CmC
2. But the `tokenName` has to match what is on Cexplorer, and now CmC?

If so, it makes it harder to automate this. I have a script 100% working, that just pulls all the info from CmC (metadata) and Dex Hunter (token logos), but if we also need to fetch the exact name, as it is displayed on Cexplorer, it adds layers of complexity as Cexplorer does not have APIs, and they do not like their data being scraped.

https://github.com/user-attachments/assets/7f87b054-1dfb-4226-bc63-8d991eaee652

Reply:
This token follows CIP-68 so its metadata is stored on-chain in the datum of UTxO containing its reference NFT. If you need an API with refined data, you can try Blockfrost, it has an API endpoint to query metadata for a specific asset: https://blockfrost.dev/api/specific-asset
![image](https://github.com/user-attachments/assets/19c7e538-972e-4e63-bd6d-55474353cb78)
CIP-68: https://cips.cardano.org/cip/CIP-68
reference NFT: https://cexplorer.io/asset/asset1v35r9yjwh4mat0t277edwnkz7tdghs4ls469ru/preview#data
![Screenshot 2024-08-21 at 15 55 22](https://github.com/user-attachments/assets/a529d7e9-ffad-4130-b4f1-e5866d000411)

Reply:
@dongntp Thanks for your reply. It is still not 100% clear to me **_what_** data we need from **_which_** source. Can we have a clear list of `source` -> `data`, please?

I can see that we require data from `CmC`, based on `CMCslug` in the `json` file, and this is where I pull all the metadata from, but then you raised an issue, saying that `tokenName` is wrong, even though that is waht `CmC` returns, but it has to be like on the `blockfrost/cexplorer`. This is the confusing part. We rely on `CmC`, but some data on it might be incorrect?

It seems that, even though we rely on `CmC`, we still need to cross-check with `blockfrost` for the field like `tokenName`? Right? Or grab all the data from the `blockfrost` and also grab the  `CMCslug` from the `CmC`? 

Some more clarity on this would really help to finish the automated script for token metadata generation.

---
1. CMCslug - from CmC
2. Rest of the metadata - blockfrost
⏫ Correct?

Reply:
Update: I will update the script to use `Blockfrost` API for all the metadata, and just grab the `CMCslug` from `CmC`